R/modified_update.R

```diff
@@ -383,19 +397,19 @@ mu_build_data <- function(con, index, meta, pop) {
   v <- if (x$activity_type == "campaign")
     "coverage_target_new" else "pop_routine"
   dat$fvps_new <- dat$coverage_new * dat[[v]]
-  i <- !is_blank(dat$coverage_new) & is_blank(dat$fvps_new)
-  if (any(i) & meta$touchstone_mod$touchstone_name != meta$group$touchstone_name[meta$group$index == index]) {
-    ## For routine these should only be MHL
+  
+  ## At the moment we can do nothing for MHL, TUV and XK, because we have no routine pop for them 
```

Comment:
Let me rerun reports locally to check.

Reply:
If you're running against the 201710gavi touchstones, the demographic data is complete for MHL, TUV, XK, just like all other countries. So dat$pop_routine should have good values for those countries, and 401-409 are lines that shouldn't happen. (If I'm understanding correctly)

Reply:
I can see the population data for these countries in 'demographic_source = 7', but 'demographic_source = 7' is not matched to any earlier touchstone yet. So dat$pop_routine is still NA for these countries.

Reply:
According to my understanding of the modified update, demographic_source is determined by any touchstone to be updated, rather than the targeting touchstone (201710gavi in this case).

Reply:
That's still fine - demographic source 7 (dds-201710) has the full demography for TUV, MHL, XK.

Reply:
The link between demographic_source and touchstone is also deprecated now - need to use the links between demographic_dataset and touchstone_demographic_dataset...

SELECT demographic_dataset from touchstone_demographic_dataset WHERE touchstone='201710gavi-2'

gives you a list of demographic_dataset ids for that touchstone. (I get 48..69 and 71)

SELECT * FROM demographic_dataset WHERE id=48   gets you a description

description = Oct/2017 ... Fertility: Births (number of) - quinquennial
demographic_statistic_type = 17
demographic_source = 7

That said, good point, I have forgotten to update the legay touchstone_demographic_source table in the recent update, which I can do if necessary - but ideally, we want to drop that table as it's no longer used in the API.

Reply:
"dds-201710" is perfect. But, I mean - for example - updating 201510gavi against 201710gavi, "unwpp_2012" is used, rather than "dds-201710".

Reply:
Oh wait, I've misunderstood you completely.

So you're saying that you need the population in the demographic sources used by the OLD touchstone, in order to do the modified update. ie - you'd need MHL, TUV, XK to exist fully in (eg) demographic source 6 (unwpp_2017_1), 3 (unwpp_2017), 2 (unwpp_2015) or 1 (unwpp_2012), which would be linked to the earlier touchstone you're updating...

Reply:
Right. :+1:

Reply:
Ok - so is this something we need to fix? ie, Do we:
1. retro-add the small countries to the older demographic datasets) OR
2. have your code handle the absence of those countries as it currently does.

Reply:
I don't think we really need to invest time for this at least for now. They are indeed small counties. Given that relevant reports need to be shipped out today or tomorrow, it is better to just flag out this issue in Nick's report.